From jirislaby@gmail.com  Mon Mar 10 14:23:36 2008
From: Jiri Slaby <jirislaby@gmail.com>
Date: Mon, 10 Mar 2008 15:36:47 +0100
Subject: USB: s2255drv, fix memset(0)
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Dean Anderson <dean@sensoray.com>, Andrew Morton <akpm@linux-foundation.org>, Jiri Slaby <jirislaby@gmail.com>
Message-ID: <1205159807-11701-1-git-send-email-jirislaby@gmail.com>

From: Jiri Slaby <jirislaby@gmail.com>

Swap second and third paramter of memset in s2255_board_init, since it's
apparently wrong. Also do not re-init its entries to 0 again.

Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/image/s2255drv.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/usb/image/s2255drv.c
+++ b/drivers/usb/image/s2255drv.c
@@ -2393,9 +2393,7 @@ static int s2255_board_init(struct s2255
 	for (j = 0; j < MAX_PIPE_BUFFERS; j++) {
 		struct s2255_pipeinfo *pipe = &dev->pipes[j];
 
-		memset(pipe, sizeof(struct s2255_pipeinfo), 0);
-		pipe->state = 0;
-		pipe->prev_state = 0;
+		memset(pipe, 0, sizeof(*pipe));
 		pipe->dev = dev;
 		pipe->cur_transfer_size = DEFAULT_PIPE_USBBLOCK;
 		pipe->max_transfer_size = MAX_PIPE_USBBLOCK;
